src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3519,7 +3519,11 @@ private void VisitTupleExpression(BoundTupleExpression node)
                     this.State[slot] = NullableAnnotation.NotNullable;
                     TrackNullableStateOfTupleElements(slot, tupleOpt, arguments, elementTypes, useRestField: false);
                 }
-                _resultType = TypeSymbolWithAnnotations.Create(tupleOpt.WithElementTypes(elementTypes), NullableAnnotation.NotNullable);
+
+                tupleOpt = tupleOpt.WithElementTypes(elementTypes);
+                var locations = tupleOpt.TupleElements.SelectAsArray(x => x.Locations.IsEmpty ? node.Syntax.Location : x.Locations[0]);
+                tupleOpt.CheckConstraints(_conversions, includeNullability: true, node.Syntax, locations, compilation, Diagnostics);
```

Comment:
minor: perhaps `CheckConstraints` call should be moved into `WithElementTypes` (by passing a `shouldCheckConstraints=true` argument to it) for consistency.

Reply:
I considered that. In looking at the code though there are a number of `WithElementX` methods in the compiler and none of them do constraint checking. It seemed more in line with current behavior to do the constraint checking in this fashion.